src/main/resources/scripts/rwa-2943_mi_data_fixes_ia.sql

```diff
@@ -0,0 +1,66 @@
+
+-- DATA CHANGE IMPLEMENTATION BY JURISDICTION - IA
+---------------------------------------------------------------
+-- PRE - CHECKS
+----------------------
+select count(*), state from tasks t  where jurisdiction  = 'IA' and created < to_timestamp('2023-09-18 16:00:00','YYYY-MM-DD HH24:MI:SS') group by state
+--285	    ASSIGNED
+--101	    CANCELLED
+--288	    COMPLETED
+--494451	TERMINATED
+--774	    UNASSIGNED
+
+select count(*), state from reportable_task rt  where jurisdiction = 'IA' and task_id in (select task_id from tasks where created < to_timestamp('2023-09-18 16:00:00','YYYY-MM-DD HH24:MI:SS'))  group by state
+--278	  ASSIGNED
+--20	  COMPLETED
+--82468	TERMINATED
+--743	  UNASSIGNED
+
+ select  count(*), state, update_action  from cft_task_db.reportable_task rt where (first_assigned_date is null or first_assigned_date_time is null or wait_time_days is null or wait_time is null or
+  completed_date is null or completed_date_time is null or processing_time_days is null or processing_time is null or  handling_time_days is null or handling_time is null or
+  due_date_to_completed_diff_days is null or due_date_to_completed_diff_time is null or is_within_sla is null or
+  role_category_label is null or jurisdiction_label is null or case_type_label is null or state_label is null or final_state_label is null)
+  and created  < to_timestamp('2023-09-18 16:00:00','YYYY-MM-DD HH24:MI:SS') and state = 'TERMINATED' and jurisdiction = 'IA' group by state, update_action ;
+--1	    TERMINATED	AutoAssign
+--1   	TERMINATED	Configure
+--33689	TERMINATED	Terminate
+--33923	TERMINATED	TerminateException
+
+ select count(*) from task_assignments ta where task_id in
+    (select task_id from tasks where created < to_timestamp('2023-09-18 16:00:00','YYYY-MM-DD HH24:MI:SS') and jurisdiction = 'IA'
+      and task_id in (select distinct task_id from task_history th  where created < to_timestamp('2023-09-18 16:00:00','YYYY-MM-DD HH24:MI:SS')))
+-- 74,795
+
+-- IMPLEMENTATION IN BATCHES
+--------------------------------
```

Comment:
batch processing grouping is by (6yrs 11months, 6months, 3months , 1month , 1 month). Is this deliberate?

Reply:
Hello Ben, we did some analysis and have it in : rwa-2829_mi_data_fix_checks_all_jurisdictions.sql (some dates to pick from), arranged these dates to chunk number of records